xapi/xapi.go

```diff
@@ -0,0 +1,143 @@
+package xapi
+
+import (
+	"io"
+	"io/ioutil"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tidb/xapi/tablecodec"
+	"github.com/pingcap/tidb/xapi/tipb"
+)
+
+// SelectResult is used to get response rows from SelectRequest.
+type SelectResult struct {
+	req      *tipb.SelectRequest
+	resp     kv.Response
+	subResp  *tipb.SelectResponse
+	cursor   int
+	finished bool
+}
+
+// Next returns the next row.
+func (r *SelectResult) Next() (handle int64, data []types.Datum, err error) {
```

Comment:
Will there be any other interface for executor to access response from multiple region servers?